core/iwasm/interpreter/wasm_loader.c

```diff
@@ -612,6 +628,22 @@ load_type_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
             type->param_cell_num = (uint16)param_cell_num;
             type->ret_cell_num = (uint16)ret_cell_num;
         }
+#if WASM_ENABLE_TYPE_POINTER_COMPARE != 0
+        /* Normalize the type list, check whehter two types in the type list
+           are equivalent, and if yes, free the later one and set it to the
+           former one */
+        for (i = 0; i < type_count - 1; i++) {
+            uint32 j;
+            for (j = i + 1; j < type_count; j++) {
+                if (module->types[j]->ref_count == 1
```

Comment:
Does L638 mean we don't normalize the type when `ref_count != 1` ?

Reply:
Should initialize the type->ref_count, set it to 1 when initializing module->types[i].

And no need to check whether wasm_type_equal here, should add the check after module->types[i] is created, if there is same type in types[0] to types[i-1], increase the latter's ref_count and destroy current type.